packages/studio-base/src/panels/Plot/builders/CurrentCustomDatasetsBuilder.ts

```diff
@@ -0,0 +1,227 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { ChartDataset } from "chart.js";
+
+import { toSec, isTime } from "@foxglove/rostime";
+import { Immutable, Time } from "@foxglove/studio";
+import { RosPath } from "@foxglove/studio-base/components/MessagePathSyntax/constants";
+import parseRosPath from "@foxglove/studio-base/components/MessagePathSyntax/parseRosPath";
+import { simpleGetMessagePathDataItems } from "@foxglove/studio-base/components/MessagePathSyntax/simpleGetMessagePathDataItems";
+import { fillInGlobalVariablesInPath } from "@foxglove/studio-base/components/MessagePathSyntax/useCachedGetMessagePathDataItems";
+import { Bounds1D } from "@foxglove/studio-base/components/TimeBasedChart/types";
+import { GlobalVariables } from "@foxglove/studio-base/hooks/useGlobalVariables";
+import { PlayerState } from "@foxglove/studio-base/players/types";
+import { getLineColor } from "@foxglove/studio-base/util/plotColors";
+
+import { CsvDataset, IDatasetsBuilder } from "./IDatasetsBuilder";
+import { Dataset } from "../ChartRenderer";
+import { Datum, isReferenceLinePlotPathType } from "../internalTypes";
+import { PlotConfig } from "../types";
+
+type DatumWithReceiveTime = Datum & {
+  receiveTime: Time;
+};
+
+type SeriesItem = {
+  enabled: boolean;
+  messagePath: string;
+  parsed: RosPath;
+  dataset: ChartDataset<"scatter", DatumWithReceiveTime[]>;
+};
+
+export class CurrentCustomDatasetsBuilder implements IDatasetsBuilder {
+  #parsedPath?: Immutable<RosPath>;
```

Comment:
seems like this is the X path - would be helpful to name it as such

```suggestion
  #parsedXPath?: Immutable<RosPath>;
```